app/src/main/java/org/p2p/wallet/user/repository/UserTokensDatabaseRepository.kt

```diff
@@ -0,0 +1,142 @@
+package org.p2p.wallet.user.repository
+
+import java.math.BigInteger
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+import org.p2p.core.crypto.Base58String
+import org.p2p.core.dispatchers.CoroutineDispatchers
+import org.p2p.core.token.Token
+import org.p2p.core.token.findByMintAddress
+import org.p2p.core.utils.fromLamports
+import org.p2p.token.service.api.events.manager.TokenServiceEvent
+import org.p2p.token.service.api.events.manager.TokenServiceEventListener
+import org.p2p.token.service.api.events.manager.TokenServiceEventManager
+import org.p2p.token.service.api.events.manager.TokenServiceEventPublisher
+import org.p2p.token.service.api.events.manager.TokenServiceEventType
+import org.p2p.token.service.model.TokenServiceNetwork
+import org.p2p.token.service.model.TokenServicePrice
+import org.p2p.wallet.home.db.TokenDao
+import org.p2p.wallet.home.model.TokenComparator
+import org.p2p.wallet.home.model.TokenConverter
+
+class UserTokensDatabaseRepository(
+    private val tokenServiceEventManager: TokenServiceEventManager,
+    private val userLocalRepository: UserLocalRepository,
+    private val tokensDao: TokenDao,
+    private val tokenConverter: TokenConverter,
+    private val coroutineDispatchers: CoroutineDispatchers,
+    private val tokenServiceEventPublisher: TokenServiceEventPublisher
+) : UserTokensLocalRepository, CoroutineScope {
```

Comment:
what's the new way of implementing CoroutineScope for repository?
i am against it, it will bring more chaos then it is